core/src/view/view.cpp

```diff
@@ -118,29 +118,46 @@ void View::roll(float _droll) {
 }
 
 void View::pitch(float _dpitch) {
-
+    m_tilted = true;
     setPitch(m_pitch + _dpitch);
 
 }
 
-void View::orbit(float _x, float _y, float _radians) {
-
-    glm::vec2 radial = { _x, _y };
-    glm::vec2 displacement = glm::rotate(radial, _radians) - radial;
-    translate(-displacement.x, -displacement.y);
-    roll(_radians);
-
+void View::undoFocusPosition(glm::vec2 _focus, glm::vec2 _prevFocus) {
+    if ( !m_focusUpdated && _prevFocus.x != 0.0f && _prevFocus.y != 0.0f) {
+        m_focusUpdated = true;
+        translate( _prevFocus.x - _focus.x, _prevFocus.y - _focus.y);
+    } else {
+        m_focusUpdated = false;
+    }
 }
 
 void View::update() {
 
+    float oldViewCenterX, oldViewCenterY, viewCenterX, viewCenterY;
+
+    oldViewCenterX = viewCenterX = m_vpWidth * 0.5;
+    oldViewCenterY = viewCenterY = m_vpHeight * 0.5;
+
     if (!m_dirty) {
         m_changed = false;
         return;
     }
 
+    // origin before tilt
+    if (m_tilted) {
+        screenToGroundPlane(oldViewCenterX, oldViewCenterY);
+    }
+
     updateMatrices();
 
+    // origin after tilt
+    if(m_tilted) {
+        screenToGroundPlane(viewCenterX, viewCenterY);
+        translate(oldViewCenterX - viewCenterX, oldViewCenterY - viewCenterY);
```

Comment:
Thats not true, updateMatrices does not modify anything in m_pos.x and m_pos.y.
The reason it has to be done here is because we need to "focus point" values before and after matrices are updated. And do necessary translations.

And if translations were compounded then you will see the focus point move a lot, which does not happen and remains consistent with the center of the view.

updateTiles is what needs updated m_pos values, and this translate happens before updateTiles.